graphium/data/datamodule.py

```diff
@@ -1252,9 +1252,17 @@ def _make_multitask_dataset(
 
         # calculate statistics for the train split and used for all splits normalization
         if stage == "train":
-            self.get_label_statistics(
-                self.processed_graph_data_path, self.data_hash, multitask_dataset, train=True
-            )
+            if self.processed_graph_data_path is not None:
+                data_path = self.processed_graph_data_path
+            elif self.cache_data_path is not None:
+                data_path = self.cache_data_path
+            else:
+                data_path = None
```

Comment:
What's the difference between processed_graph_data_path and cache_data_path here?

Reply:
When `cache_data_path` is set, the entire dataset is loaded into memory. If `processed_graph_data_path` is set, each graph is read from disk. In the particular function you marked, previously only the `processed_graph_data_path` was passed to obtain the label statistics. However, if one were to use `cache_data_path` (which is feasible for smaller datasets such as ToyMix), then without the above change, the label statistics are always re-calculated.

Reply:
As discussed today, let's remove `cache_data_path`, and always use `processed_graph_data_path`. When the reading is from RAM, simply pre-load everything into memory before training.

Also, don't forget to delete the old cache.

Reply:
> Also, don't forget to delete the old cache.

I'm not sure I understand what you mean here.